ui/core/src/api/SifService/SifService.ts

```diff
@@ -129,50 +142,51 @@ export default function createSifService({
       return supportedTokens;
     },
 
-    async connect() {
-      const keplrProvider = await keplrProviderPromise;
+    async setClient() {
+      if (!offlineSigner) {
+        client = null;
+        return;
+      }
+      const accounts = await offlineSigner.getAccounts();
+      const address = accounts.length > 0 ? accounts[0].address : "";
+      if (!address) {
+        throw "No address on sif account";
+      }
+      client = new SifClient(
```

Comment:
This doesn't seem to me as if it really needs to be exposed as public API - would there be benefit from a refactorability perspective to making this simply a pure function? 

offlineSigner -> client

Reply:
I made a refactoring to remove offline signer from exposed vars, but the client should be exposed, at least with current setup. tests are written in such way that they initialise client from mnemonic and then set this exposed client var to it. Can be refactored for sure, but will take more time and I really want to push it asap so it gets into deployment